File: projet/client_leger/polyquiz/lib/models/quiz_interface.dart
import 'package:polyquiz/enums/question_type.dart';

class Quiz {
  final String id;
  final String title;
  final String description;
  final int duration;
  final String? lastModification;
  final List<QuizQuestion> questions;
  final bool? visible;

  Quiz({
    required this.id,
    required this.title,
    required this.description,
    required this.duration,
    this.lastModification,
    required this.questions,
    this.visible,
  });
}

class QuizQuestion {
  final QuestionType type;
  final String text;
  final int points;
  final List<QuizChoice>? choices;

  QuizQuestion({
    required this.type,
    required this.text,
    required this.points,
    this.choices,
  });
}

class QuizChoice {
  final String text;
  final bool? isCorrect;

  QuizChoice({
    required this.text,
    this.isCorrect,
  });
}

File: projet/common/interfaces/quiz.interface.ts
import { QuestionType } from '../enums/question-type.enum';

export interface Quiz {
    id: string;
    title: string;
    description: string;
    duration: number;
    lastModification: string | null;
    questions: QuizQuestion[];
    visible?: boolean;
    owner?: string; //firebase uid string
}

export interface QuizQuestion {
    type: QuestionType;
    text: string;
    points: number;
    choices?: QuizChoice[]; //Only QCM
    answer? : number, //Only QRE
    interval? : {max:number, min:number} //Only QRE
    margin?: number //Only QRE
    imageUrl?: string;
}

export interface QuizChoice {
    text: string;
    isCorrect?: boolean | null;
}
